### ISSUE TITLE
Set Gateway reference with {namespace}/{name} format on VirtualService

### ISSUE THREAD

**Issue by piotrmsc** [kyma-project/kyma#8175]
**Description**

API Gateway should use new format for referencing gateways.

Currently it sets FQDN like here :
```
apiVersion: v1
items:
- apiVersion: gateway.kyma-project.io/v1alpha1
  kind: APIRule
  metadata:
    creationTimestamp: "2020-04-23T09:31:56Z"
    generation: 1
    name: test
    namespace: pm
    resourceVersion: "142191"
    selfLink: /apis/gateway.kyma-project.io/v1alpha1/namespaces/pm/apirules/test
    uid: 667b30f5-e665-4e7a-838e-fe8004651901
  spec:
    gateway: kyma-gateway.kyma-system.svc.cluster.local
    rules:
    - accessStrategies:
      - config: {}
        handler: noop
      methods:
      - GET
      - POST
      - PUT
      - DELETE
      path: /.*
    service:
      host: pm.xxx.com
      name: httpbin
      port: 8000
  status:
    APIRuleStatus:
      code: OK
    accessRuleStatus:
      code: OK
    lastProcessedTime: "2020-04-23T09:31:56Z"
    observedGeneration: 1
    virtualServiceStatus:
      code: OK
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

 but should use {namespace}/{gateway_name} like this : 

``` 
gateways:
  - some-config-namespace/my-gateway
```

Reference : https://istio.io/docs/reference/config/networking/gateway/

For backward compatibility, controller should allow both formats of gateway field on API CR.
CBS queries may need to be updated and we will need to update our test definitions.

**Reasons**
Align with new istio patterns

**Attachments**

**Comment by piotrmsc** [comment]
We need to do it probably asap after #9474

**Comment by Demonsthere** [comment]
New PR for istio 1.7: https://github.com/kyma-project/kyma/pull/9503

**Comment by elchead** [comment]
I tried to update to the new format but it broke the tests (see PR)

**Comment by strekm** [comment]
@cnvergence please check gateway regex. i think there filed limitation saying at least 3 chars, which is wrong. please update also this.

**Comment by mjakobczyk** [comment]
Testing PR that uses following regex: `^[0-9a-z-_]+((\/|\.)[0-9a-z-_]+)*$`:
- tested using https://regex101.com site
- it still allows names such as: `name/name/name` (should be invalid)
As discussed on Friday, instead of introducing one regex we could instead offer two - the one handling backward compatibility and the second one serving a new `{namespace}/{gateway_name}` approach. In the second one we could be strict on how many limiters (such as `/`) are used to avoid positive evaluation of a mentioned phrase.

**Comment by strekm** [comment]
please adjust [Expose a service](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-api-exposure/apix-02-expose-service-apigateway/) and [Expose and secure a service](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-api-exposure/apix-03-expose-and-secure-service/) tutorials with this naming convention
